File: pack/TreeNode.java
package pack;

public class TreeNode {
  public int val;
  public TreeNode left;
  public TreeNode right;
  public TreeNode() {}
  public TreeNode(int val) { this.val = val; }
  public TreeNode(int val, TreeNode left, TreeNode right) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
  
  public void printTree() {
    if( val != 0) printTree(this);
    System.out.println();
  }
  public void printTree( TreeNode root) {
    System.out.print(root.val);
    if( root.left != null) printTree( root.left);
    if( root.right != null) printTree( root.right);
  }

  public TreeNode populateTree( Object[] input) {
    populateTree(this, 0, input);
    return this;
  }
  public void populateTree( TreeNode root, int index, Object[] input ) {
    if( input.length > index && input[index] != null) root.val = (int) input[index];
    if( input.length > ((index * 2) + 1) && input[((index * 2) + 1)] != null) { 
      root.left = new TreeNode(); 
      populateTree(root.left, ((index * 2) + 1), input); 
    }
    if( input.length > ((index * 2) + 2) && input[((index * 2) + 2)] != null) { 
      root.right = new TreeNode(); 
      populateTree(root.right, ((index * 2) + 2), input); 
    }
  }
}

File: pack/TreeNode.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=1, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

    def populateFromScratch(self, n): # Binary Search Tree (1 to n)
        if n >= (self.val * 2):     self.left = TreeNode(self.val*2).populateFromScratch(n)
        if n >= (self.val * 2) + 1: self.right = TreeNode(self.val * 2 + 1).populateFromScratch(n)
        return self
    
    def populateFromArr(self, arr, i = 0): # Binary Tree (filled according to arr)
        if len(arr) > i: self.val = arr[i]
        if len(arr) > (((i+1)*2)-1): self.left = TreeNode(None).populateFromArr(arr, (((i+1)*2)-1))
        if len(arr) > (((i+1)*2)): self.right = TreeNode(None).populateFromArr(arr, (((i+1)*2)))
        return self

    def traverse(self):
        print(self.val, end = ' ')
        if self.left: self.left.traverse()
        if self.right: self.right.traverse()